src/Storages/MergeTree/ReplicatedMergeTreeSink.cpp

```diff
@@ -224,10 +226,11 @@ void ReplicatedMergeTreeSink::consume(Chunk chunk)
         streams += temp_part.streams.size();
         if (streams > max_insert_delayed_streams_for_parallel_write)
         {
-            finishDelayedChunk(zookeeper);
+            finishDelayedChunk();
```

Comment:
What we expect in `delayed_chunk->zookeeper` inside first call of `finishDelayedChunk`? Should it already be set? Previously we passed current session, but not it's assigned to delayed_chunk after this call

Reply:
Either `delayed_chunk` is not set too (so `finishDelayedChunk` will just exit) or both `delayed_chunk` and `delayed_chunk->zookeeper` were set on prev iteration